sdcm/tester.py

```diff
@@ -883,71 +884,97 @@ def setUp(self):  # pylint: disable=too-many-branches,too-many-statements
+                            change_default_password(node=db_cluster.nodes[0],
+                                                    user=self.params.get('authenticator_user'),
+                                                    password=self.params.get('authenticator_password'))
+                            # TODO: Replace with strong password generation
+                else:
                     self.set_system_auth_rf(db_cluster=db_cluster)
 
-                    self._setup_ldap_roles(db_cluster=db_cluster)
-                    if self.params.get('ldap_server_type') == LdapServerType.MS_AD:
-                        change_default_password(node=db_cluster.nodes[0],
-                                                user=self.params.get('authenticator_user'),
-                                                password=self.params.get('authenticator_password'))
-                        # TODO: update proper loader and monitor in multi-tenant case.
-                        #       Now it updates only the first ones all the time.
-                        self.loaders.added_password_suffix = True
-                        # TODO: Replace with strong password generation
-                        self.monitors.added_password_suffix = True
-            else:
-                self.set_system_auth_rf(db_cluster=db_cluster)
-
-            db_node_address = db_cluster.nodes[0].ip_address
+        def _create_loaders():
             if self.loaders and not self.loaders_multitenant:
                 self.loaders_multitenant = [self.loaders]
             for loaders in self.loaders_multitenant:
                 if loaders:
-                    loaders.wait_for_init(db_node_address=db_node_address)
+                    loaders.wait_for_init()
+
+        def _create_gemini_cluster():
+            # cs_db_cluster is created in case MIXED_CLUSTER. For example, gemini test
+            if self.cs_db_cluster:
+                self.init_nodes(db_cluster=self.cs_db_cluster)
+
+        def _create_monitors():
+            if self.monitors and not self.monitors_multitenant:
+                self.monitors_multitenant = [self.monitors]
+
+            if self.monitors_multitenant:
+                monitors_init_in_parallel = ParallelObject(
+                    timeout=3600,
+                    objects=[[monitor] for monitor in self.monitors_multitenant],
+                    num_workers=len(self.monitors_multitenant))
+                monitors_init_in_parallel.run(
+                    func=(lambda m: m.wait_for_init()),
+                    unpack_objects=True, ignore_exceptions=False)
+
+        def _validate_raft():
+            for db_cluster in self.db_clusters_multitenant:
+                if db_cluster:
+                    db_cluster.validate_seeds_on_all_nodes()
+                    validate_raft_on_nodes(nodes=db_cluster.nodes)
+                    db_cluster.start_kms_key_rotation_thread()
```

Comment:
This is not related to raft

And it starts a thread of its own, no point putting it into a thread

But anyhow the title of the function is misleading

Reply:
name fixed. thread moved out.